File: src/main/java/com/example/demo/DemoApplication.java
package com.example.demo;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.web.bind.annotation.*;

@SpringBootApplication
public class DemoApplication {
    public static void main(String[] args) {
        SpringApplication.run(DemoApplication.class, args);
    }
}

@RestController
class PostController {
    private int i = 0;

    @GetMapping("/")
    public String all() {
        i += 1;
        return String.format("Hello, world! (%d)", i);
    }
}

File: src/main/kotlin/com/example/demo/DemoApplication.kt
package com.example.demo

import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication
import org.springframework.web.bind.annotation.RestController
import org.springframework.web.bind.annotation.GetMapping

@SpringBootApplication
class DemoApplication

fun main() {
    runApplication<DemoApplication>()
}

@RestController
class PostController {
    var i = 0;

    @GetMapping("/")
    fun all(): String {
        i += 1
        return "Hello, world! $i"
    }
}
